server/services/messageServices.js

```diff
@@ -175,35 +178,38 @@ export default class MessageService {
         messageId: newMessage.id,
       });
     }
+    this.messages.push(newMessage);
     return newMessage;
   }
 
   getMessageById(id) {
-    const message = this.AllMessage()[id - 1];
+    const foundMessage = this.AllMessage().find(element => element.id === Number(id));
     if (!id) {
```

Comment:
This check should come before you use the `id` to look for the message in the list of messages.

Reply:
Done